hermes/src/store/storage.rs

```diff
@@ -134,8 +134,20 @@ pub trait Storage: Send + Sync {
         filter: MessageStateFilter,
     ) -> Result<Vec<MessageState>>;
 
+    /// Store the accumulator state. Please note that this call will replace the
+    /// existing accumulator state for the given state's slot. If you wish to
+    /// update the accumulator state, use `update_accumulator_state` instead.
     async fn store_accumulator_state(&self, state: AccumulatorState) -> Result<()>;
```

Comment:
Couple thoughts on this change:
1. is there ever a situation where it's safe to use this `store` function? Maybe you'd be better off deleting it?
2. Does the same change need to be done to storing the message_states above? I think the answer is "no", because you never mutate the content of `MessageState`. But then it's a little weird that the two types of stored values require different handling. You could make this consistent by splitting out the two components of AccumulatorState and fetching/storing them separately. That would be easy to do if you created a cache abstraction that handled the insertion / eviction logic (which would be good anyway, because i see that code is repeated for messages and accumulator states below).

Reply:
I agree with you and David suggested the same. I will do it in another PR. having a proper abstraction for the simple ordered cache makes sense too.